api/src/main/java/com/nicknathanjustin/streamercontracts/users/UserServiceImpl.java

```diff
@@ -35,4 +46,34 @@ public void login(@NonNull final UserModel user) {
         user.setLastLogin(loginTimestamp);
         userModelRepository.save(user);
     }
+
+    @Override
+    public TwitchUser getTwitchUserFromUsername(@NonNull String twitchUsername) {
+        final HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.set("Client-ID", clientId);
+        final HttpEntity<String> entity = new HttpEntity<String>("parameters", headers);
+
+        final RestTemplate restTemplate = new RestTemplate();
+        final String url = userInfoUri + "?login=" + twitchUsername;
```

Comment:
This is fine for now but it might be useful to make this into a function. We'll probably need to make more REST calls to Twitch and other services soon.

Reply:
Yeah, we can maybe do something similar as the FE where we wrote a REST client class.